static/js/components/email/lib_test.js

```diff
@@ -260,6 +261,43 @@ describe('Specific email config', () => {
           "value": "Paid"
         }]);
       });
+
+      describe('getFilters', () => {
+        it('should return filters', () => {
+          let filters = getFilters(JSON.parse(queryFilters));
+          assert.deepEqual(filters, [{
+            "id": "program.enrollments.payment_status",
+            "name": "program.enrollments.payment_status",
+            "value": "Paid"
+          }]);
+        });
+      });
+
+      describe('findTerms', () => {
+        it('should return filters', () => {
+          let filters = findTerms(JSON.parse(queryFilters));
+          assert.equal(filters.length, 1);
+          assert.deepEqual(filters, [{
+            "program.enrollments.payment_status": "Paid"
+          }]);
+        });
+
+        it('should ignore program.id and return empty list when no filter selected', () => {
+          const query = `{
+            "bool": {
+              "must":[
+                {
+                  "term": {
+                    "program.id":1
+                  }
+                }
+              ]
+            }
+          }`;
+          let filters = findTerms(JSON.parse(query));
+          assert.equal(filters.length, 0);
+        });
+      });
```

Comment:
I would suggest that you try to come up with tests that more methodically cover all the possible cases for what the input to `findTerms` could look like and what the output should be - just to start it should cover these at a minimum:

```js
it('should return an empty list if passed an empty object', ...);

it('should return an empty list if passed an object without any "term" key on it', ...);

it('should return an empty list if passed an object without any nested "term" keys', ...);

it('should return a list of filters defined at the top-level', ...);

it('should return a list of filters which are nested in the object', ...);

it('should return a list of filters which are nested at different levels', ...);
```

there are probably a few more key cases to test, but yeah, I think if we write some good unit tests for the function we can have a lot of confidence that it's implemented correctly.